src/main/kotlin/nl/info/client/brp/BrpClientService.kt

```diff
@@ -54,22 +64,26 @@ class BrpClientService @Inject constructor(
 
     fun queryPersonen(personenQuery: PersonenQuery): PersonenQueryResponse =
         updateQuery(personenQuery).let {
-            personenApi.personen(it)
+            personenApi.personen(
+                personenQuery = it,
+                purpose = purposeSearch.validatePurpose("brp.doelbinding.zoekmet")
+            )
         }
 
     /**
      * Retrieves a person by burgerservicenummer from the BRP Personen API.
      *
      * @param burgerservicenummer the burgerservicenummer of the person to retrieve
      * @return the person if found, otherwise null
+     *
+     * @throws BrpInvalidPurposeException if the configured purpose is invalid or missing.
      */
-    fun retrievePersoon(burgerservicenummer: String): Persoon? =
-        (
-            personenApi.personen(
-                createRaadpleegMetBurgerservicenummerQuery(burgerservicenummer)
-            ) as RaadpleegMetBurgerservicenummerResponse
-            )
-            .personen?.firstOrNull()
+    fun retrievePersoon(burgerservicenummer: String): Persoon? = (
+        personenApi.personen(
+            personenQuery = createRaadpleegMetBurgerservicenummerQuery(burgerservicenummer),
+            purpose = purposeRetrieve.validatePurpose("brp.doelbinding.raadpleegmet")
```

Comment:
Not using the constant here:
```suggestion
            purpose = purposeRetrieve.validatePurpose(ENV_VAR_BRP_DOELBINDING_RAADPLEEGMET)
```